src/main/java/net/fabricmc/loader/game/MinecraftGameProvider.java

```diff
@@ -164,6 +164,15 @@ public EntrypointTransformer getEntrypointTransformer() {
 		return TRANSFORMER;
 	}
 
+	@Override
+	public boolean canOpenErrorGui() {
+		if (arguments == null || envType == EnvType.CLIENT) {
+			return true;
+		}
+		List<String> extras = arguments.getExtraArgs();
+		return !extras.contains("nogui") && !extras.contains("--nogui");
```

Comment:
When I tested the server crashed so I don't think so. (However making it case sensitive *might* make sense?)

Reply:
I mean more like making it case insensitive if mc allows non-lowercase and it hasn't already been lowercased. It is possible that the two cases are being parsed&handled differently. This would probably have to happen in the argument handler, not your PR.